polyfactory/factories/base.py

```diff
@@ -630,6 +636,11 @@ def get_field_value(  # noqa: C901, PLR0911, PLR0912
         if field_build_parameters is None and cls.should_set_none_value(field_meta=field_meta):
             return None
 
+        if cls.should_use_default_value(field_meta):
+            if callable(field_meta.default):
```

Comment:
Does this logic hold for types that are callable? For example,

```
@dataclass
class A:
    type_field: type = field(default=int)
    callable: Callable[[], Any] = field(default=lambda: 'hello world')
```
would produce `A(0, 'hello world')`

Reply:
I'm thinking of scrapping this approach that I took. I think the easier way would be to just do the `should_use_defalt_value` check in `process_kwargs` and if we should be using the default value, then just not include the field in the `kwargs` that we then give to the model constructor. This way, we don't have to worry about any of these edge cases. I think this will allow us to not worry about the defaults in `sqlalchemy` that you had mentioned. Thoughts?

Reply:
To clarify, if `should_use_default_value` returns `True`, then we never call `get_field_value`.

Reply:
This sounds great me. It probably catches some other cases too where the models themselves set defaults.

On the value being return a similar thing is needed in my other PR. I used `Ignore` as the return to check for this but `Null` might be more appropriate. I think it'll need a similar check as done here https://github.com/litestar-org/polyfactory/pull/468/files#diff-1b4e816a1af7b945e990030b74241cc33c80c0b585a42daf5e41ae55f684e00dR898. What do you think?

Reply:
Yupp, it makes things far easier for `polyfactory`. Also, yeah, I think `Null` would be more appropriate in your use case as well.